python/pyspark/sql/tests/pandas/test_pandas_udf_scalar.py

```diff
@@ -748,6 +750,31 @@ def check_vectorized_udf_return_scalar(self):
             with self.assertRaisesRegex(Exception, "Return.*type.*Series"):
                 df.select(f(col("id"))).collect()
 
+    def test_udf_with_variant_input(self):
+        df = self.spark.range(0, 10).selectExpr("parse_json(cast(id as string)) v")
+        from pyspark.sql.functions import col
+
+        scalar_f = pandas_udf(lambda u: str(u), StringType())
+        iter_f = pandas_udf(
+            lambda it: map(lambda u: str(u), it), StringType(), PandasUDFType.SCALAR_ITER
+        )
+        expectedErrorStr = 'UDFs do not support "VARIANT" type input data'
+        for f in [scalar_f, iter_f]:
+            with self.assertRaisesRegex(AnalysisException, expectedErrorStr):
+                df.select(f(col("v"))).collect()
```

Comment:
Can we check the error class instead of the error string here?

Reply:
done!